tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/DAGImpl.java

```diff
@@ -1772,6 +1773,17 @@ private static void parseVertexEdges(DAGImpl dag, Map<String, EdgePlan> edgePlan
 
     vertex.setInputVertices(inVertices);
     vertex.setOutputVertices(outVertices);
+    boolean cleanupShuffleDataAtVertexLevel = dag.dagConf.getInt(TezConfiguration.TEZ_AM_VERTEX_CLEANUP_HEIGHT,
+        TezConfiguration.TEZ_AM_VERTEX_CLEANUP_HEIGHT_DEFAULT) > 0 && ShuffleUtils.isTezShuffleHandler(dag.dagConf);
+    if (cleanupShuffleDataAtVertexLevel) {
+      int deletionHeight = dag.dagConf.getInt(TezConfiguration.TEZ_AM_VERTEX_CLEANUP_HEIGHT,
+              TezConfiguration.TEZ_AM_VERTEX_CLEANUP_HEIGHT_DEFAULT);
+
+      VertexShuffleDataDeletionContext vShuffleDeletionContext =
```

Comment:
what about putting all this to VertexImpl, like: 
```
vertex.initShuffleDeletionContext(deletionHeight)
```
?
that way you don't have to expose the init logic to DagImpl

Reply:
ack. Will resolve in next revision.